osgar/lib/virtual_bumper.py

```diff
@@ -5,17 +5,21 @@
 """
 import math
 
+from osgar.lib.mathex import normalizeAnglePIPI
 
-def equal_poses(pose1, pose2, dist_limit):
+
+def equal_poses(pose1, pose2, dist_limit, angle_limit=None):
     x1, y1, heading1 = pose1
     x2, y2, heading2 = pose2
-    return math.hypot(x2 - x1, y2 - y1) < dist_limit
+    return (math.hypot(x2 - x1, y2 - y1) < dist_limit and 
+            (angle_limit is None or abs(normalizeAnglePIPI(heading1 - heading2)) < angle_limit))
```

Comment:
There is no differentiation between linear and angular move. If the update_desired_speed indicates only angular direction should change but we only see linear position changing, it should still trigger bumper (as we have non-zero angular speed but heading is not moving), yet virtual bumper is happy enough with some movement.

Reply:
what values do you use for limit angle? I expected something  between 45-90deg used primarily for turn-in-place. Smaller and larger values probably do not have much sense.

Reply:
I would say an angle that identifies turning is happening as opposed the robot is stuck, ie anything beyond the margin of error/noise.  Using it with pi/16 now. If the turn is impeded (eg by rubbing against an obstacle), it may be quite slow but happening nevertheless so no bumper needed.